### PR TITLE
Implement `srand`

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: https://github.com/lfortran/lfortran/issues/4593

### CODE REVIEW

**Comment by certik** [comment]
The error is:
```
2024-09-29T07:51:42.6914721Z [ 42%] Linking Fortran executable intrinsics_129
2024-09-29T07:51:42.6932611Z wasm-ld: warning: function signature mismatch: _lfortran_init_random_seed
2024-09-29T07:51:42.6938201Z >>> defined as (i32) -> i32 in CMakeFiles/intrinsics_126.dir/intrinsics_126.f90.o
2024-09-29T07:51:42.6940762Z >>> defined as (i32) -> void in /home/runner/work/lfortran/lfortran/src/bin/../runtime/lfortran_runtime_wasm_wasi.o
```

File: src/libasr/pass/intrinsic_subroutines.h [hunk 1]
```diff
@@ -150,6 +151,57 @@ namespace RandomSeed {
...
+    static inline ASR::asr_t* create_Srand(Allocator& al, const Location& loc, Vec<ASR::expr_t*>& args, diag::Diagnostics& /*diag*/) {
+        Vec<ASR::expr_t*> m_args; m_args.reserve(al, 1); m_args.push_back(al, args[0]);
+        return ASR::make_IntrinsicImpureSubroutine_t(al, loc, static_cast<int64_t>(IntrinsicImpureSubroutines::Srand), m_args.p, m_args.n, 0);
+    }
+
+    static inline ASR::stmt_t* instantiate_Srand(Allocator &al, const Location &loc,
+            SymbolTable *scope, Vec<ASR::ttype_t*>& arg_types,
+            Vec<ASR::call_arg_t>& new_args, int64_t /*overload_id*/) {
+
+        std::string c_func_name = "_lfortran_init_random_seed";
```
Comment (certik) [inline src/libasr/pass/intrinsic_subroutines.h:172]: What is the definition of this function : `int -> void` or `int -> int`?

Comment (certik) [reply src/libasr/pass/intrinsic_subroutines.h:172]: If it is `int->int`, then it needs to be changed to `int -> void`, then then it should be correct, and also WASM will work.

Comment (HarshitaKalani) [reply src/libasr/pass/intrinsic_subroutines.h:172]:
```
LFORTRAN_API void _lfortran_init_random_seed(unsigned seed)
{
    srand(seed);
}
```
Yeah, It is already `int -> void`.

**Comment by HarshitaKalani** [comment]
```
 385/1093 Test  #386: intrinsics_126 ...........................***Failed    0.25 sec
Error: failed to run main module `/home/runner/work/lfortran/lfortran/integration_tests/test-llvm_wasm/intrinsics_126`

Caused by:
    0: failed to invoke command default
    1: error while executing at wasm backtrace:
           0:  0x304 - <unknown>!signature_mismatch:_lfortran_init_random_seed
           1:  0x36e - <unknown>!_lcompilers_srand_
           2:  0x403 - <unknown>!main
           3:  0x51d - <unknown>!_start
           4: 0x140d8 - <unknown>!_start.command_export
       note: using the `WASMTIME_BACKTRACE_DETAILS=1` environment variable may show more debugging information
    2: wasm trap: wasm `unreachable` instruction executed
 ```

File: src/libasr/pass/intrinsic_subroutines.h (hunk 1 above)
Comment (certik) [reply src/libasr/pass/intrinsic_subroutines.h:172]: This is the runtime, where it is `int -> void`. But what signature is the instantiated code that calls this? Is it calling it as `int->int` and ignoring the result?

Comment (HarshitaKalani) [reply src/libasr/pass/intrinsic_subroutines.h:172]: Can I change the runtime signature to `int -> int`? That seemed simple to me.

**Review by certik — APPROVED** [review]
Thanks for getting it working!

Later we can improve this.

In a subsequent PR, do you think you could please add a warning that this function is an "LFortran extension", and recommend using "random_init` instead?

**Comment by certik** [comment]
Go ahead and resolve the conflicts, then merge it as is. The other improvements we can do later.

**Pull request merged**